src/inmanta/util/dict_path.py

```diff
@@ -489,6 +542,131 @@ def get_elements(self, container: object) -> list[object]:
             and all(any(key.matches(k) and value.matches(v) for k, v in dct.items()) for key, value in self.key_value_pairs)
         ]
 
+    def resolve_wild_cards(self, container: object) -> Sequence["WildKeyedList"]:
+        """
+        Get one WildKeyedList by set of matching key value pairs that can be found
+        in the given container at the relation specified in this object.
+
+        ..code-block:: python
+
+            assert WildKeyedList("relation","key_attribute","*").resolve_wild_cards(
+                {
+                    "relation":[
+                        {
+                            "key_attribute":"key_value",
+                            "other_attribute":"other_value",
+                        },
+                        {
+                            "key_attribute":"other_value",
+                        },
+                        {
+                            "other_key_attribute":"other_value",
+                        },
+                    [
+                }
+            ) == [
+                WildKeyedList("relation", "key_attribute", "key_value"),
+                WildKeyedList("relation", "key_attribute", "other_value"),
+            ]
+
+            assert WildKeyedList("relation",[("key_attribute","*"), ("other_attribute", "*")]).resolve_wild_cards(
+                {
+                    "relation":[
+                        {
+                            "key_attribute":"key_value",
+                            "other_attribute":"other_value",
+                        },
+                        {
+                            "key_attribute":"other_value",
+                        },
+                        {
+                            "other_key_attribute":"other_value",
+                        },
+                    [
+                }
+            ) == [
+                WildKeyedList("relation", [("key_attribute","key_value"), ("other_attribute", "other_value")]),
+            ]
+
+            assert WildKeyedList("relation",[("*","other_value")]).resolve_wild_cards(
+                {
+                    "relation":[
+                        {
+                            "key_attribute":"key_value",
+                            "other_attribute":"other_value",
+                        },
+                        {
+                            "key_attribute":"other_value",
+                        },
+                        {
+                            "other_key_attribute":"other_value",
+                        },
+                    [
+                }
+            ) == [
+                WildKeyedList("relation", [("other_attribute","other_value")]),
+                WildKeyedList("relation", [("key_attribute","other_value")]),
+                WildKeyedList("relation", [("other_key_attribute","other_value")]),
+            ]
+
+            assert WildKeyedList("relation",[("*","*")]).resolve_wild_cards(
```

Comment:
What about

```
items= WildKeyedList("relation", [("*", "other_value")]).resolve_wild_cards(
        {
            "relation": [
                {
                    "key_attribute": "key_value",
                    "other_attribute": "other_value",
                },
                {
                    "key_attribute": "other_value",
                    "key_attribute_2": "other_value",
                },
                {
                    "other_key_attribute": "other_value",
                }
                ]
        }
    )
```

Reply:
It will give 
```
[
    WildKeyedList("relation", [("other_attribute", "other_value"]),
    WildKeyedList("relation", [("key_attribute", "other_value"]),
    WildKeyedList("relation", [("key_attribute_2", "other_value"]),
    WildKeyedList("relation", [("other_key_attribute", "other_value"]),
]
```

Reply:
We can have more dict path than there are matching elements (and so more than one dict path by matching element).  Do you think that is an issue?

Reply:
And also, we can have more than one matching element by dict path (as the identity/uniqueness is not checked)